### ISSUE TITLE
Some images ignored in DICOM JP2K sample slides

### ISSUE THREAD

**Issue by bgilbert** [openslide/openslide#462]
### Operating system

Fedora 38

### Platform

x86_64

### OpenSlide version

Git main

### Slide format

DICOM

### Issue details

The JP2K [sample slides](https://openslide.cs.cmu.edu/download/openslide-testdata/DICOM/) (`CMU-1-JP2K-33005` and `JP2K-33003-1`), which were converted from Aperio slides with `com.pixelmed.convert.TIFFToDicom`, have a few issues:

1. OpenSlide ignores the highest-resolution level and all the associated images because it expects their `ImageType`s to be `ORIGINAL`, but in this case they're `DERIVED`.
1. It also ignores the Aperio thumbnail, which has an image flavor of `THUMBNAIL`.  I'm not seeing that flavor defined [here](https://dicom.innolitics.com/ciods/vl-whole-slide-microscopy-image/whole-slide-microscopy-image/00080008), though.
1. If modified to accept the overview or thumbnail images, OpenSlide fails with `Not a JPEG file: starts with 0x19 0x00`.  I haven't debugged this.

Should we fix these?  I assume it's reasonable to accept `DERIVED` images when reading converted data (especially since we silently accept only the lower-resolution levels right now).  Not sure about the others.

cc @jcupitt

**Comment by jcupitt** [comment]
1. I added `"DERIVED", "PRIMARY", "VOLUME", "NONE"` as another level type, which should fix this.
2. I've had DICOMs where the thumbnail has really been the LOCALIZER, which is not really the same. But this one seems to be a useful thumbnail, you're right, so we should add it. I've included that in this PR.
3. The thumbnail is 1024 x 732 pixels, so 1024 * 732 * 3 == 22487804 byes, which is the size of the object in the pixeldata. I'd guess this was uncompressed RGB and has been tagged incorrectly as JPEG. I've mailed david to ask about this.
